MekHQ/src/mekhq/campaign/stratcon/StratconScenarioFactory.java

```diff
@@ -0,0 +1,189 @@
+/*
+* MegaMek - Copyright (C) 2020 - The MegaMek Team
+*
+* This program is free software; you can redistribute it and/or modify it under
+* the terms of the GNU General Public License as published by the Free Software
+* Foundation; either version 2 of the License, or (at your option) any later
+* version.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+* FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+* details.
+*/
+
+package mekhq.campaign.stratcon;
+
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import megamek.common.UnitType;
+import mekhq.campaign.mission.ScenarioForceTemplate;
+import mekhq.campaign.mission.ScenarioMapParameters.MapLocation;
+import mekhq.MekHQ;
+import mekhq.MekHqConstants;
+import mekhq.Utilities;
+import mekhq.campaign.mission.ScenarioTemplate;
+import mekhq.campaign.mission.atb.AtBScenarioManifest;
+
+/**
+ * This class handles functionality related to loading and sorting scenario templates.
+ * @author NickAragua
+ */
+public class StratconScenarioFactory {
+    // loaded dynamic scenario templates, sorted by location (ground, low atmosphere, space)
+    private static Map<MapLocation, List<ScenarioTemplate>> dynamicScenarioLocationMap = new HashMap<>();
+    private static Map<Integer, List<ScenarioTemplate>> dynamicScenarioUnitTypeMap = new HashMap<>();
+    private static Map<String, ScenarioTemplate> dynamicScenarioNameMap = new HashMap<>();
+    private static String HOSTILE_FACILITY_SCENARIO = "Hostile Facility.xml";
+    private static String ALLIED_FACILITY_SCENARIO = "Allied Facility.xml";
+    
```

Comment:
As before, let's move these to MekHqConstants just to keep a single place with filenames.